src/main/java/com/example/demo/controllers/rest/ConstraintController.java

```diff
@@ -0,0 +1,60 @@
+package com.example.demo.controllers.rest;
+
+
+import com.example.demo.db.entities.Constraint;
+import com.example.demo.db.entities.ConstraintType;
+import com.example.demo.db.repo.ConstraintRepo;
+import com.example.demo.db.repo.ConstraintTypeRepo;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+@RestController
+@RequestMapping(path = "/constraint")
+public class ConstraintController extends RestApiAbstract<Constraint, ConstraintRepo,Integer>{
+
+      public ConstraintRepo getRepo() {
+        return null;
+    }
```

Comment:
Lint you text `CMD+option+L` on mac if you need for windows I can help you find

Reply:
? when needs to be done? did not understand that

Reply:
On every file you should lint...
And second the get repo shouldn't be null